src/cppcache/include/geode/Struct.hpp

```diff
@@ -138,7 +138,7 @@ class CPPCACHE_EXPORT Struct : public Serializable {
    * Returns the name of the field corresponding to the index number in the
    * Struct
    */
-  virtual const char* getFieldName(const int32_t index) const;
+  virtual const std::string* getFieldName(const int32_t index) const;
```

Comment:
Any reason why you're returning `std::string*`  instead of `std::string` or `std::string&`?